scripts/aae-export.py

```diff
@@ -1,175 +1,510 @@
+
+        aae += "Anchor Point\n"
+        aae += "\tFrame\tX pixels\tY pixels\tZ pixels\n"
+        aae += positions
+
+        aae += "\n\n"
+        aae += "Position\n"
+        aae += "\tFrame\tX pixels\tY pixels\tZ pixels\n"
+        aae += positions
+
+        aae += "\n\n"
+        aae += "Scale\n"
+        aae += "\tFrame\tX percent\tY percent\tZ percent\n"
+        aae += scales
+
+        aae += "\n\n"
+        aae += "Rotation\n"
+        aae += "\tFrame Degrees\n"
+        aae += rotations
+
+        def generate_power_pin(clip, track):
+            power_pin = ""
+
+            frames = []
+            corners = []
+            for marker in track.markers[1:-1]:
+                if not 0 < marker.frame <= clip.frame_duration:
+                    continue
+                if marker.mute:
+                    continue
+
+                frames.append(marker.frame)
+                corners.append([list(c) for c in marker.corners])
+            
+            for pini, corneri in [(2, 3), (3, 2), (4, 0), (5, 1)]:
+                power_pin += f"\n\nEffects\tCC Power Pin #1\tCC Power Pin-000{pini}"
+                power_pin += "\n\tFrame\tX pixels\tY pixels"
+
+                for i, plane in enumerate(corners):
+                    corner = plane[corneri]
+                    x = corner[0] * clip.size[0]
+                    y = (1 - corner[1]) * clip.size[1]
+                    power_pin += f"\n\t{frames[i]}\t{x:.3f}\t{y:.3f}"
+
+            return power_pin
+    
+        if do_includes_power_pin and track.markers[0].__class__.__name__ == "MovieTrackingPlaneMarker":
+            aae += generate_power_pin(clip, track)
+
+        aae += "\n\nEnd of Keyframe Data\n"
+
+        return aae
+
+    @staticmethod
+    def _export_to_file(clip, track, aae, prefix, do_do_not_overwrite):
+        """
+        Parameters
+        ----------
+        clip : bpy.types.MovieClip
+        track : bpy.types.MovieTrackingTrack or MovieTrackingPlaneTrack
+        aae : str
+            Likely coming from _generated().
+        prefix : None or str
+        do_do_not_overwrite : bool
+            AAEExportSettings.do_do_not_overwrite.
+
+        """
+        coords = None
+        if track.markers[0].__class__.__name__ == "MovieTrackingMarker":
+            for marker in track.markers:
+                if not marker.mute:
+                    coords = (marker.co[0] * clip.size[0], (1 - marker.co[1]) * clip.size[1])
+                    break
+        else: # "MovieTrackingPlaneMarker"
+            for marker in track.markers:
+                if not marker.mute:
+                    coords = AAEExportExportAll._plane_track_center(marker.corners[0], marker.corners[1], marker.corners[2], marker.corners[3])
+                    coords = (coords[0] * clip.size[0], (1 - coords[1]) * clip.size[1])
+                    break
+
+        if coords != None:
+            p = Path(clip.filepath if not prefix else prefix)
```

Comment:
The paths here can be relative paths in Blender syntax (started with `//` and relative to the current project file's location), so they need to be made into absolute paths using `bpy.path.abspath()`.

Reply:
A `.blend` project file can contain many individual video clips. You can track a lot of clips without the need to close Blender or create a new Blender project.  
![image](https://user-images.githubusercontent.com/112813970/202879792-105b93cd-eb13-4ec2-8077-a1b5bcbf8ba8.png)

That's why the export file is not relative to the `.blend` project file, but relative to individual video files.  
![image](https://user-images.githubusercontent.com/112813970/202879833-fa135ffa-98ef-46ea-b8a2-1e942b6f476f.png)

And also that's where the awkward code below is coming from.  
![image](https://user-images.githubusercontent.com/112813970/202879854-aa79f4cd-6890-4542-b241-0681d8505f48.png)

Reply:
I agree, but what I meant is that this caused an error for me on Windows since `clip.filepath` was of the form `//path/to/subdirectory/clip.mp4`, which is blender's syntax for a path relative to the project file. I'm not sure if that's always the case, but it was for me. So I meant converting the clip's file path to an absolute path that can be passed to `open` with
```py
p = Path(bpy.path.abspath(clip.filepath if not prefix else prefix))`
```

Reply:
Oh I didn't know `clip.filepath` could be relative. https://github.com/Akatmks/Akatsumekusa-Aegisub-Scripts/commit/0c9a6c1aa2f1723c245367e69a0c58978975fa20  

Btw if you know the solution, you can add it to the sciprt yourself. You can either commit it directly to main, or create a new branch and merge it to main, whichever method you prefer.